clifford/_multivector.py

```diff
@@ -707,14 +707,18 @@ def _pick_inv(self, fallback):
         ----------
         fallback : bool, optional
             If `None`, perform no checks on whether normal inv is appropriate.
-            If `True`, fallback to a linalg approach if necessary.
+            If `True`bn , fallback to a Hitzer and Sanguines method if possible and a linalg approach if not.
             If `False`, raise an error if normal inv is not appropriate.
         """
         Madjoint = ~self
         MadjointM = (Madjoint * self)
         if fallback is not None and not MadjointM.isScalar():
             if fallback:
-                return self.leftLaInv()
+                nbases = sum(self.layout.sig != 0)
+                if nbases <= 5:
+                    return self.hitzer_inverse()
+                else:
+                    return self.leftLaInv()
```

Comment:
To avoid computing the sum twice, perhaps:
```suggestion
                try:
                    return self.hitzer_inverse()
                except NotImplementedError:
                    return self.leftLaInv()
```